runner/lib/main.js

```diff
@@ -582,8 +643,21 @@ const main = async (progName, rawArgs, powers) => {
             });
 
             await runLoadgenResult.updateConfig(null);
-            stageConsole.log('Waiting for loadgen tasks to end');
-            await orInterrupt(Promise.race([idle, sleep(2 * 60 * 1000)]));
+            const maxLoadgenDuration =
+              Object.values(stats.cycles).reduce(
+                (max, cycleStats) =>
+                  cycleStats &&
+                  cycleStats.duration != null &&
```

Comment:
I don't think it is worth it. Please expand to test both `null` and `undefined` if that's the semantics you want. My reasoning:

Experienced JS and Node programmers will know the idiom. To any readers both unfamiliar with the idiom and with JS's truly bizarre `==` coercion rules, this is a barrier to understanding. One of my favorite Carl Hewitt quotes:

   ***A program should not only work, it should appear to work***

This code does not say what it means. Code should be read more often than it is written. The expertise required to read a piece of code and basically puzzle it out should be less than the expertise needed to write that code.

If you did want to keep the idiom, I'd say you must annotate each occurrence with an explanatory comment, which kinda defeats the purpose.

Reply:
Thanks for bringing this to my attention.